Implement a sample using jQuery + ES6.

crsanti opened the pull request:
This closes #54.

crsanti committed:
Implemented jQuery sample.
Corrected pattern validator to validate empty values.

crsanti committed:
Renamed sample

luisgonzalo commented on lib/src/rules/email.ts:
```diff
@@ -1,26 +1,15 @@
 
   validationResult.succeeded = isValid;
   validationResult.type = 'EMAIL';
   validationResult.errorMessage = isValid ? '' : 'Please enter a valid email address.';
```
Move the message 'Please enter a valid email address' to a resources file or class for easy translation?

crsanti replied:
It would be a good idea for i18n. It could be public and be accesed via `Validators.messages.EMAIL_PATTERN` or something like that. I'll wait for some feedback.

brauliodiez replied:
The error message provided by the validationResult is just "courtesy", is not in the scope of the library or validators to handle multilanguage (the library would do too much). To handle multilangue there is another filed that "type" this field informs you the type of validation that has failed, there you can implement an ngMessage + Multilanguage approach (outside from the validator itself), similar approach as angular.

I think what @luisgonzalo is pointing is a good idea to open a new issue: implement multilanguage sample, @luisgonzalo would you like to be volunteer for implementing this case? we will guide you.

luisgonzalo replied:
Sure, I would like to implement it with your guidance :)

luisgonzalo commented on samples/jquery/00 ShoppingForm/src/modules/app/app.js:
```diff
@@ -0,0 +1,152 @@
+  onBrandSelect(event) {
+    this.onFieldChange(event);
+    const product = Number(event.currentTarget.value) || null;
+    const isValidProduct = product !== null;
```
(product !== null)

crsanti replied:
Parenthesis is not required.

luisgonzalo replied:
I know, it was only to improve readability a bit ;)

brauliodiez replied:
Here we should stick to an standard, personally I preffer the parthensis, easier to read, but I let the team to decide

luisgonzalo commented on samples/jquery/00 ShoppingForm/src/modules/app/app.js:
```diff
@@ -0,0 +1,152 @@
+    if (!isValidProduct) {
+      $selProducts.val('');
+    }
+    this.toggleAvailability($selProducts, product !== null);
```
this.toggleAvailability($selProducts, isValidProduct)

crsanti replied:
Yep! I'll fix that!

luisgonzalo commented on samples/jquery/00 ShoppingForm/src/modules/app/validation/rules/nifValidator.js:
```diff
@@ -0,0 +1,45 @@
+  const validationResult = new FieldValidationResult();
+
+  validationResult.succeeded = isValid;
+  validationResult.errorMessage = isValid ? '' : 'The given value is not a valid NIF/NIE';
```
Take this message to a resources file or class for translation?

crsanti replied:
I think this is not necessary for a sample. I did just keep it simple.

brauliodiez replied:
Same as before, errorMessage is just friendly sugar, "type" is the real thing

luisgonzalo commented on samples/react/01 SignupForm/src/components/sampleSignupForm/validations/signupFormValidation.ts:
```diff
@@ -3,15 +3,13 @@ import {
 function passwordAndConfirmPasswordValidationHandler(value: any, vm: any): FieldValidationResult {
   const passwordAndConfirmPasswordAreEqual = vm.password === value;
-  const errorInfo: string = (passwordAndConfirmPasswordAreEqual) ? '' : 'Passwords do not match';
+  const errorInfo = (passwordAndConfirmPasswordAreEqual) ? '' : 'Passwords do not match';
```
move message to translation class?

crsanti committed:
Merge branch 'master' into issue#54-jquery-sample

crsanti committed:
Corrected duplicated comparation.

nasdan commented on samples/jquery/00 ShoppingForm/src/css/site.css:
```diff
@@ -0,0 +1,21 @@
+  border-radius: 0 !important;
+}
+
+/*React apply activeClassName to <a> element, but Bootstrap active class is over <li> element*/
```
Remove comment?

nasdan commented on samples/jquery/00 ShoppingForm/src/modules/app/app.js:
```diff
@@ -0,0 +1,152 @@
+
+  loadProductsByBrand(id) {
+    if (id !== null) {
+      const osList = this.getProductsByBrand(id);
```
what is `osList`?

nasdan commented on samples/jquery/00 ShoppingForm/src/modules/app/app.js:
```diff
@@ -0,0 +1,152 @@
+
+  getProductsByBrand(id) {
+    const brand = this.brands.filter(product => product.id === id);
+    return brand[0] && brand[0].products;
```
This is a bit weird. Does it return products if `brand[0]` is not Undefined?

crsanti replied:
Yes. If `brand[0]` does not exists because the filter didn't match it will return `undefined`. It's like:

```js
if(brand[0]) {
  return brand[0].products;
}
```

The point of this is to get `product[] | undefined` and pass it to `loadSelect($select, items = []` (it will initialize to `[]` if `items` is `undefined`.

nasdan commented on samples/jquery/00 ShoppingForm/src/modules/app/app.js:
```diff
@@ -0,0 +1,152 @@
+  };
+
+  handleFieldValidationResult($form) {
+    return (function (fieldValidationResult) {
```
Why don't use here fat arrow `() =>`?

nasdan commented on samples/jquery/00 ShoppingForm/src/modules/app/validation/rules/nifValidator.js:
```diff
@@ -0,0 +1,45 @@
+    const letter = valueToUpper.substr(-1);
+    const charIndex = parseInt(nie.substr(0, 8)) % 23;
+
+    if (validChars.charAt(charIndex) === letter) {
```
you can write this code like:

```
isValid = validChars.charAt(charIndex) === letter;
```

nasdan commented on samples/jquery/00 ShoppingForm/webpack.config.js:
```diff
@@ -0,0 +1,92 @@
+
+module.exports = {
+  context: path.join(basePath, "src"),
+  resolve: {
```
If you want you can remove `resolve` section because [webpack](https://webpack.js.org/configuration/resolve/#resolve-extensions) has as default extensions `.js and .json`

nasdan commented on samples/jquery/00 ShoppingForm/webpack.config.js:
```diff
@@ -0,0 +1,92 @@
+  plugins: [
+    new webpack.optimize.CommonsChunkPlugin({
+      name: 'vendor',
+      filaneme: 'vendor.js'
```
As other PR #59 , remove this line

nasdan commented on samples/jquery/00 ShoppingForm/src/modules/app/validation/formProductValidationService.js:
```diff
@@ -0,0 +1,44 @@
+    product: [
+      {
+        validator: Validators.required,
+        eventsFilter: { onSubmit: true, onChange: true },
```
I think this line is not necessary because you are not using `onSubmit` eventFilter to filter one field.
In this [sample](https://github.com/Lemoncode/lcFormValidation/pull/60/files#diff-d7eb573276fa017c19ae6370d857b28aR90), you are using  `validateForm` to validate all fields at `onSubmit` event.

And  about `onChange` is the default filter, at this point as you want. If you want highlight that you can use eventsFilter, it's ok for me

brauliodiez commented:
@crsanti give us a buzz when this is ready for a third review.

crsanti commented:
I'll work on this right now (just fixed some typos in doc page)

brauliodiez commented:
right on !!! we are almost there

crsanti committed:
Applying feedback

Removed css file.
Replaced `products` with `brands` in service.
Some minor changes.

crsanti committed:
More updates

Removed eventsFilter in formValidation.
Removed unnecessary code from webpack.config.js.
Changed event from `change` to `keyup` for better experience.

crsanti commented:
I think this is ready to review! :+1:

brauliodiez commented:
Reviewed and merged. just only missing the ES6 version of the samples :)
